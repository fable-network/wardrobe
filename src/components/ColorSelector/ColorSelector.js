import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const Wrapper = styled('span')`
  display: inline-flex;
  align-items: center;
  cursor: ${(p) => (p.onClick ? 'pointer' : 'initial')};
`;

const fixedSizeCss = css`
  width: ${(p) => p.fixedSize};
  height: ${(p) => p.fixedSize};
`;

const missingColorCss = css`
  position: relative;

  &::after {
    position: absolute;
    content: "?";
    text-align: center;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    font-weight: 500;
  }
`;

const selectedCss = css`
  transform: scale(1) !important;
`;

const ColorCircle = styled('span')`
  background-color: ${(p) => p.theme.white};
  border-radius: 100%;
  box-shadow: 0px 0px 0px 1px #cecece;
  border: 2px solid ${(p) => p.theme.white};
  transform: scale(.8);
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  transition: all .3s ease;
  background-repeat: no-repeat;
  background-position: 50% 50%;

  &:hover {
    transform: scale(.9);
  }

  ${(p) => (p.fixedSize ? fixedSizeCss : 'padding: .75em;')}
  ${(p) => (p.isMissingColor ? missingColorCss : null)}
  ${(p) => (p.disableInteraction || p.selected ? selectedCss : null)}
  ${(p) => (
    p.patternImage
      ? `background-image: url(${p.patternImage})`
      : `background: ${p.color || p.theme.white}`
  )}
`;

const Text = styled('span')`
  margin-left: 8px;
  font-weight: ${(p) => (p.selected ? '500' : '400')};
`;

const ColorSelector = props => {
  const {
    onClick,
    patternImage,
    color,
    disableInteraction,
    text,
    selected,
    fixedSize,
    ...otherProps
  } = props;
  const isMissingColor = !patternImage && !color;

  return (
    <Wrapper {...otherProps} onClick={!disableInteraction ? onClick : null}>
      <ColorCircle
        color={color}
        selected={selected}
        fixedSize={fixedSize}
        patternImage={patternImage}
        isMissingColor={isMissingColor}
        disableInteraction={disableInteraction}
      />
      {text && <Text selected={selected}>{text}</Text>}
    </Wrapper>
  );
};

ColorSelector.defaultProps = {
  selected: false,
  disableInteraction: false,
};

ColorSelector.propTypes = {
  color: PropTypes.string,
  /** overrides color if available */
  patternImage: PropTypes.string,
  onClick: PropTypes.func,
  /** ignores click handler if true */
  disableInteraction: PropTypes.bool,
  text: PropTypes.string,
  selected: PropTypes.bool,
  /** overrides dynamic size based on font-size */
  fixedSize: PropTypes.string,
};

export default ColorSelector;
