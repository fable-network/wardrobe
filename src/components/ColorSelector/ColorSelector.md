Color selector (without interaction):

```jsx
<ColorSelector
  color="#e25454"
  disableInteraction={true}
/>
```

Color Selector with click handler:

```jsx
<ColorSelector
  color="#5F9DC7"
  onClick={() => alert('Clicked!')}
/>
```

Color selector with pattern image:

```jsx
<ColorSelector
  patternImage="src/static/images/patternImage.jpg"
/>
```

Color Selector with text:

```jsx
<ColorSelector
  color="#aecc76"
  text="Lime green"
/>
```

Selected color selector:

```jsx
<ColorSelector
  color="#aecc76"
  selected={true}
  text="Lime green"
/>
```

As part of a group:

```jsx
<div>
  <ColorSelector
    color="#5F9DC7"
    onClick={() => alert('#5F9DC7')}
    text="Sky blue"
  />
</div>
<div>
  <ColorSelector
    color="#aecc76"
    text="Lime green"
    onClick={() => alert('#aecc76')}
  />
</div>
<div>
  <ColorSelector
    color="#e25454"
    disableInteraction={true}
    text="Flame red (Selected)"
    onClick={() => alert('#e25454')}
    selected={true}
  />
</div>
<div>
  <ColorSelector
    patternImage="src/static/images/patternImage.jpg"
    onClick={() => alert('Pattern image')}
    text="Pattern image"
  />
</div>
