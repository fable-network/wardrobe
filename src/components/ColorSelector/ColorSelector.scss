//
// ColorSelector
//
@import '../../globals/scss/colors';
@import '../../globals/scss/typography';
@import '../../globals/scss/variables';
@import '../../globals/scss/import-once';

@include exports('colorSelector') {
  .#{$prefix}--colorSelector {
    
    &--colorCircle {
      background-color: #fff;
      border-radius: 100%;
      box-shadow: 0px 0px 0px 1px #cecece;
      border: 2px solid #FFF;
      transform: scale(.8);
      display: inline-block;
      vertical-align: middle;
      overflow: hidden;
      padding: 1em;
      transition: all .3s ease;
      background-repeat: no-repeat;
      background-position: 50% 50%;
    }

    &:hover,
    &--state--noInteraction,
    &--state--selected {
      .#{$prefix}--colorSelector--colorCircle {
        transform: scale(1);
      }
    }

    &--hasPointerCursor {
      cursor: pointer;
    }

    &--text {
      vertical-align: middle;
      margin-left: 5px;
      &--state--selected {
        font-weight: 500;
      }
    }
  }
}
