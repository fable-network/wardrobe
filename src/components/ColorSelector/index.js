export default from './ColorSelector.js'
