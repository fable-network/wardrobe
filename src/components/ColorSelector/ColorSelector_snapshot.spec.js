/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import ColorSelector from './ColorSelector';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('ColorSelector Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <ColorSelector
          color="#e25454"
          disableInteraction={true}
        />
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = mount(
        <ColorSelector
          color="#5F9DC7"
          onClick={() => alert('Clicked!')}
        />
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = mount(
        <ColorSelector
          patternImage="images/patternImage.jpg"
        />
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = mount(
        <ColorSelector
          color="#aecc76"
          text="Lime green"
        />
      );
      expect(wrapper4).toMatchSnapshot();

      const wrapper5 = mount(
        <ColorSelector
          color="#aecc76"
          selected={true}
          text="Lime green"
        />
      );
      expect(wrapper5).toMatchSnapshot();

      const wrapper6 = mount(
        <ColorSelector
          color="#e25454"
          fixedSize='40px'
          onClick={() => alert('40px')}
        />
      );
      expect(wrapper6).toMatchSnapshot();

      const wrapper7 = mount(
        <ColorSelector />
      );
      expect(wrapper7).toMatchSnapshot();

    });
  });
});
