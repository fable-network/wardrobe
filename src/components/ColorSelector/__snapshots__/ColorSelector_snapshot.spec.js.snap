// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    color="#e25454"
    disableInteraction={true}
    selected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          color="#e25454"
          disableInteraction={true}
          isMissingColor={false}
          selected={false}
        />,
        undefined,
      ],
      "onClick": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#e25454",
          "disableInteraction": true,
          "fixedSize": undefined,
          "isMissingColor": false,
          "patternImage": undefined,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            color="#e25454"
            disableInteraction={true}
            isMissingColor={false}
            selected={false}
          />,
          undefined,
        ],
        "onClick": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#e25454",
            "disableInteraction": true,
            "fixedSize": undefined,
            "isMissingColor": false,
            "patternImage": undefined,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ColorSelector Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    color="#5F9DC7"
    disableInteraction={false}
    onClick={[Function]}
    selected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          color="#5F9DC7"
          disableInteraction={false}
          isMissingColor={false}
          selected={false}
        />,
        undefined,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#5F9DC7",
          "disableInteraction": false,
          "fixedSize": undefined,
          "isMissingColor": false,
          "patternImage": undefined,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            color="#5F9DC7"
            disableInteraction={false}
            isMissingColor={false}
            selected={false}
          />,
          undefined,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#5F9DC7",
            "disableInteraction": false,
            "fixedSize": undefined,
            "isMissingColor": false,
            "patternImage": undefined,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ColorSelector Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    disableInteraction={false}
    patternImage="images/patternImage.jpg"
    selected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          disableInteraction={false}
          isMissingColor={false}
          patternImage="images/patternImage.jpg"
          selected={false}
        />,
        undefined,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": undefined,
          "disableInteraction": false,
          "fixedSize": undefined,
          "isMissingColor": false,
          "patternImage": "images/patternImage.jpg",
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            disableInteraction={false}
            isMissingColor={false}
            patternImage="images/patternImage.jpg"
            selected={false}
          />,
          undefined,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "disableInteraction": false,
            "fixedSize": undefined,
            "isMissingColor": false,
            "patternImage": "images/patternImage.jpg",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ColorSelector Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    color="#aecc76"
    disableInteraction={false}
    selected={false}
    text="Lime green"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          color="#aecc76"
          disableInteraction={false}
          isMissingColor={false}
          selected={false}
        />,
        <ColorSelector__Text
          selected={false}
        >
          Lime green
        </ColorSelector__Text>,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#aecc76",
          "disableInteraction": false,
          "fixedSize": undefined,
          "isMissingColor": false,
          "patternImage": undefined,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lime green",
          "selected": false,
        },
        "ref": null,
        "rendered": "Lime green",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            color="#aecc76"
            disableInteraction={false}
            isMissingColor={false}
            selected={false}
          />,
          <ColorSelector__Text
            selected={false}
          >
            Lime green
          </ColorSelector__Text>,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#aecc76",
            "disableInteraction": false,
            "fixedSize": undefined,
            "isMissingColor": false,
            "patternImage": undefined,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lime green",
            "selected": false,
          },
          "ref": null,
          "rendered": "Lime green",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ColorSelector Component snapshots should match snapshot 5`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    color="#aecc76"
    disableInteraction={false}
    selected={true}
    text="Lime green"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          color="#aecc76"
          disableInteraction={false}
          isMissingColor={false}
          selected={true}
        />,
        <ColorSelector__Text
          selected={true}
        >
          Lime green
        </ColorSelector__Text>,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#aecc76",
          "disableInteraction": false,
          "fixedSize": undefined,
          "isMissingColor": false,
          "patternImage": undefined,
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lime green",
          "selected": true,
        },
        "ref": null,
        "rendered": "Lime green",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            color="#aecc76"
            disableInteraction={false}
            isMissingColor={false}
            selected={true}
          />,
          <ColorSelector__Text
            selected={true}
          >
            Lime green
          </ColorSelector__Text>,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#aecc76",
            "disableInteraction": false,
            "fixedSize": undefined,
            "isMissingColor": false,
            "patternImage": undefined,
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lime green",
            "selected": true,
          },
          "ref": null,
          "rendered": "Lime green",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ColorSelector Component snapshots should match snapshot 6`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    color="#e25454"
    disableInteraction={false}
    fixedSize="40px"
    onClick={[Function]}
    selected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          color="#e25454"
          disableInteraction={false}
          fixedSize="40px"
          isMissingColor={false}
          selected={false}
        />,
        undefined,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#e25454",
          "disableInteraction": false,
          "fixedSize": "40px",
          "isMissingColor": false,
          "patternImage": undefined,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            color="#e25454"
            disableInteraction={false}
            fixedSize="40px"
            isMissingColor={false}
            selected={false}
          />,
          undefined,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#e25454",
            "disableInteraction": false,
            "fixedSize": "40px",
            "isMissingColor": false,
            "patternImage": undefined,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ColorSelector Component snapshots should match snapshot 7`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSelector
    disableInteraction={false}
    selected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ColorSelector__ColorCircle
          disableInteraction={false}
          isMissingColor={true}
          selected={false}
        />,
        undefined,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": undefined,
          "disableInteraction": false,
          "fixedSize": undefined,
          "isMissingColor": true,
          "patternImage": undefined,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ColorSelector__ColorCircle
            disableInteraction={false}
            isMissingColor={true}
            selected={false}
          />,
          undefined,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "disableInteraction": false,
            "fixedSize": undefined,
            "isMissingColor": true,
            "patternImage": undefined,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
